Update `svg/animations` tests to use local `SVGTestCase` and leverage `js-test.js`

Ahmad-S792 opened the pull request:
#### 0e14d4f96ced357830f7d96e360724fc8a561d86
<pre>
Update `svg/animations` tests to use local `SVGTestCase` and leverage `js-test.js`

<a href="https://bugs.webkit.org/show_bug.cgi?id=268763">https://bugs.webkit.org/show_bug.cgi?id=268763</a>

Reviewed by Nikolas Zimmermann.

This patch aims to leverage `js-test.js` in `svg/animations` by updating helper
script &apos;SVGTestCase.js&apos;.
Similar to Blink, in this patch, we separate &apos;SVGTestCase&apos; from dynamic-updates
and add it as local resources in &apos;svg/animations/resources&apos; folder. It is up-to-date
copy from Blink / Chromium source but I removed unused function to slim it down.

&gt; Script Update:
* LayoutTests/svg/animations/resources/SVGTestCase.js:

&gt; Tests Update:
* LayoutTests/svg/animations/animate-color-rgba-calcMode-discrete.html:
* LayoutTests/svg/animations/animate-elem-03-t-drt.html:
* LayoutTests/svg/animations/animate-elem-04-t-drt.html:
* LayoutTests/svg/animations/animate-elem-05-t-drt.html:
* LayoutTests/svg/animations/animate-elem-06-t-drt.html:
* LayoutTests/svg/animations/animate-elem-07-t-drt.html:
* LayoutTests/svg/animations/animate-elem-08-t-drt.html:
* LayoutTests/svg/animations/animate-elem-09-t-drt.html:
* LayoutTests/svg/animations/animate-elem-10-t-drt.html:
* LayoutTests/svg/animations/animate-elem-11-t-drt.html:
* LayoutTests/svg/animations/animate-elem-12-t-drt.html:
* LayoutTests/svg/animations/animate-elem-13-t-drt.html:
* LayoutTests/svg/animations/animate-end-attribute.html:
* LayoutTests/svg/animations/animate-endElement-beginElement.html:
* LayoutTests/svg/animations/animate-marker-orient-from-angle-to-autostartreverse.html:
* LayoutTests/svg/animations/animate-mpath-insert.html:
* LayoutTests/svg/animations/animate-path-animation-cC-sS-inverse.html:
* LayoutTests/svg/animations/animate-path-animation-Cc-Ss.html:
* LayoutTests/svg/animations/animate-path-animation-lL-vV-hH-inverse.html:
* LayoutTests/svg/animations/animate-path-animation-Ll-Vv-Hh.html:
* LayoutTests/svg/animations/animate-path-animation-qQ-tT-inverse.html:
* LayoutTests/svg/animations/animate-path-animation-Qq-Tt.html:
* LayoutTests/svg/animations/animate-path-nested-transforms.html:
* LayoutTests/svg/animations/animate-path-to-animation.html:
* LayoutTests/svg/animations/change-baseVal-while-animating-fill-freeze-2.html:
* LayoutTests/svg/animations/change-baseVal-while-animating-fill-remove-2.html:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-freeze.html:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-remove.html:
* LayoutTests/svg/animations/deferred-insertion.html:
* LayoutTests/svg/animations/fill-remove-support.html:
* LayoutTests/svg/animations/svglength-additive-by-5.html:
* LayoutTests/svg/animations/svglength-animation-px-to-exs.html:
* LayoutTests/svg/animations/svglength-animation-px-to-percentage.html:
* LayoutTests/svg/animations/svgpath-animation-1.html:
* LayoutTests/svg/animations/svgPreserveAspectRatio-animation-1.html:
* LayoutTests/svg/animations/use-animate-transform-and-position.html:
* LayoutTests/svg/animations/use-animate-width-and-height.html:

&gt; Updated Test Expectations (due to `js-test.js&apos;):
* LayoutTests/svg/animations/use-animate-width-and-height-expected.txt:
* LayoutTests/svg/animations/deferred-insertion-expected.txt:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-remove-expected.txt:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-freeze-expected.txt:
* LayoutTests/svg/animations/animate-color-rgba-calcMode-discrete-expected.txt:

Canonical link: <a href="https://commits.webkit.org/274447@main">https://commits.webkit.org/274447@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/962ebcde30ace0adae8586cb9faed56eb0962ccb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39154 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18114 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41507 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41688 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34871 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/20983 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15462 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/32769 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39728 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/15252 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/33937 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/13249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/13214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/34875 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42965 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/35548 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/35204 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/39034 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/13966 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/11522 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/37265 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15572 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/34153 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/15235 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5112 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/15058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann reviewed (approved):
r=me - makes sense - I also like the expected.txt changes.
Nitpick, fix typo in the commit message: `js-test.js`.

Ahmad-S792 committed:
Update `svg/animations` tests to use local `SVGTestCase` and leverage `js-test.js`

https://bugs.webkit.org/show_bug.cgi?id=268763

Reviewed by Nikolas Zimmermann.

This patch aims to leverage `js-test.js` in `svg/animations` by updating helper
script 'SVGTestCase.js'.
Similar to Blink, in this patch, we separate 'SVGTestCase' from dynamic-updates
and add it as local resources in 'svg/animations/resources' folder. It is up-to-date
copy from Blink / Chromium source but I removed unused function to slim it down.

> Script Update:
* LayoutTests/svg/animations/resources/SVGTestCase.js:

> Tests Update:
* LayoutTests/svg/animations/animate-color-rgba-calcMode-discrete.html:
* LayoutTests/svg/animations/animate-elem-03-t-drt.html:
* LayoutTests/svg/animations/animate-elem-04-t-drt.html:
* LayoutTests/svg/animations/animate-elem-05-t-drt.html:
* LayoutTests/svg/animations/animate-elem-06-t-drt.html:
* LayoutTests/svg/animations/animate-elem-07-t-drt.html:
* LayoutTests/svg/animations/animate-elem-08-t-drt.html:
* LayoutTests/svg/animations/animate-elem-09-t-drt.html:
* LayoutTests/svg/animations/animate-elem-10-t-drt.html:
* LayoutTests/svg/animations/animate-elem-11-t-drt.html:
* LayoutTests/svg/animations/animate-elem-12-t-drt.html:
* LayoutTests/svg/animations/animate-elem-13-t-drt.html:
* LayoutTests/svg/animations/animate-end-attribute.html:
* LayoutTests/svg/animations/animate-endElement-beginElement.html:
* LayoutTests/svg/animations/animate-marker-orient-from-angle-to-autostartreverse.html:
* LayoutTests/svg/animations/animate-mpath-insert.html:
* LayoutTests/svg/animations/animate-path-animation-cC-sS-inverse.html:
* LayoutTests/svg/animations/animate-path-animation-Cc-Ss.html:
* LayoutTests/svg/animations/animate-path-animation-lL-vV-hH-inverse.html:
* LayoutTests/svg/animations/animate-path-animation-Ll-Vv-Hh.html:
* LayoutTests/svg/animations/animate-path-animation-qQ-tT-inverse.html:
* LayoutTests/svg/animations/animate-path-animation-Qq-Tt.html:
* LayoutTests/svg/animations/animate-path-nested-transforms.html:
* LayoutTests/svg/animations/animate-path-to-animation.html:
* LayoutTests/svg/animations/change-baseVal-while-animating-fill-freeze-2.html:
* LayoutTests/svg/animations/change-baseVal-while-animating-fill-remove-2.html:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-freeze.html:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-remove.html:
* LayoutTests/svg/animations/deferred-insertion.html:
* LayoutTests/svg/animations/fill-remove-support.html:
* LayoutTests/svg/animations/svglength-additive-by-5.html:
* LayoutTests/svg/animations/svglength-animation-px-to-exs.html:
* LayoutTests/svg/animations/svglength-animation-px-to-percentage.html:
* LayoutTests/svg/animations/svgpath-animation-1.html:
* LayoutTests/svg/animations/svgPreserveAspectRatio-animation-1.html:
* LayoutTests/svg/animations/use-animate-transform-and-position.html:
* LayoutTests/svg/animations/use-animate-width-and-height.html:

> Updated Test Expectations (due to `js-test.js'):
* LayoutTests/svg/animations/use-animate-width-and-height-expected.txt:
* LayoutTests/svg/animations/deferred-insertion-expected.txt:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-remove-expected.txt:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-freeze-expected.txt:
* LayoutTests/svg/animations/animate-color-rgba-calcMode-discrete-expected.txt:

Canonical link: https://commits.webkit.org/274447@main
